harmony_regridding_service/regridding_service.py

```diff
@@ -889,20 +888,184 @@ def _get_vertical_dims(dims: Iterable[str], var_info: VarInfoFromNetCDF4) -> str
 
 
 def _compute_source_swath(
-    grid_dimensions: tuple[str, str], filepath: str, var_info: VarInfoFromNetCDF4
+    grid_dimensions: tuple[str, str],
+    filepath: str,
+    var_info: VarInfoFromNetCDF4,
+    variable_set: set,
 ) -> SwathDefinition:
-    """Return a SwathDefinition for the input gridDimensions."""
-    longitudes, latitudes = _compute_horizontal_source_grids(
-        grid_dimensions, filepath, var_info
-    )
+    """Return a SwathDefinition for the input grid_dimensions."""
+    if _dims_are_lon_lat(grid_dimensions, var_info):
+        longitudes, latitudes = _compute_horizontal_source_grids(
+            grid_dimensions, filepath, var_info
+        )
+    elif _dims_are_projected_x_y(grid_dimensions, var_info):
+        longitudes, latitudes = _compute_projected_horizontal_source_grids(
+            grid_dimensions, filepath, var_info, variable_set
+        )
+    else:
+        raise SourceDataError(
```

Comment:
This is the first real change to the flow and we're just adding the ability to compute a swathDefinition from projected y,x coordinates.